rpcs3/Emu/Cell/lv2/sys_rsx.cpp

```diff
@@ -251,12 +265,17 @@ error_code sys_rsx_context_iomap(u32 context_id, u32 io, u32 ea, u32 size, u64 f
 {
 	sys_rsx.warning("sys_rsx_context_iomap(context_id=0x%x, io=0x%x, ea=0x%x, size=0x%x, flags=0x%llx)", context_id, io, ea, size, flags);
 
+	const auto render = rsx::get_current_renderer();
+
 	if (!size || io & 0xFFFFF || ea + u64{size} > rsx::constants::local_mem_base || ea & 0xFFFFF || size & 0xFFFFF ||
-		context_id != 0x55555555 || rsx::get_current_renderer()->main_mem_size < io + u64{size})
+		context_id != 0x55555555 || render->main_mem_size < io + u64{size})
 	{
 		return CELL_EINVAL;
 	}
 
+	// Mapping must be atomic, enforce that by pausing rsx
```

Comment:
This I'm very unsure of. There is no hardware pause/unpause on realhw which means the driver would/should manually pause RSX by itself by issuing a semaphore op. I don't see how it would be atomic in nature without a manual pause/unpause using fifo commands, which fw cellGcmSys should already do anyway.

Reply:
You are right, it came from a misunderstading of mine in lv1.

Reply:
It can however be added to HLE gcm if its not done already.

Reply:
I dont think its a libgcm job here, I guess on realhw the whole function is not atomic from rsx view.